torch/csrc/autograd/VariableTypeUtils.h

```diff
@@ -58,6 +58,26 @@ inline void check_inplace(const at::Tensor& tensor, bool requires_grad) {
   }
 }
 
+// Same as check_inplace but returns false instead of raising an error
+inline bool can_mutate_inplace(const at::Tensor& tensor, bool requires_grad) {
```

Comment:
You could make a `std::optional<std::string> _can_mutate_inplace` to avoid duplicating the logic.

Reply:
`handle_view_on_rebase` can throw many different errors or none at all, so I don't think it's worth it. At that point you'd be as well doing `try-catch` around `check_inplace`.

Reply:
hmmm still feels bad to duplicate imo

The code change doesn't seem too bad since there's only a single `TORCH_CHECK` call we need to convert to a `return msg` at the very end, but maybe I'm missing something?

The comment above the call to `handle_view_on_rebase` in `check_inplace` is out of date btw, we don't need to worry about warns anymore.